streams/src/commonTest/kotlin/com/mirego/trikot/streams/reactive/processors/ConcatProcessorTests.kt

```diff
@@ -0,0 +1,91 @@
+package com.mirego.trikot.streams.reactive.processors
+
+import com.mirego.trikot.streams.reactive.BehaviorSubject
+import com.mirego.trikot.streams.reactive.PublisherResultAccumulator
+import com.mirego.trikot.streams.reactive.Publishers
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class ConcatProcessorTests {
+    lateinit var firstPublisher: BehaviorSubject<String>
+    lateinit var secondPublisher: BehaviorSubject<String>
+    lateinit var processor: ConcatProcessor<String>
+    lateinit var accumulator: PublisherResultAccumulator<String>
+
+    @BeforeTest
+    fun setup() {
+        firstPublisher = Publishers.behaviorSubject()
+        secondPublisher = Publishers.behaviorSubject()
+        processor = ConcatProcessor(firstPublisher, secondPublisher)
+        accumulator = PublisherResultAccumulator(processor)
+    }
+
+    @Test
+    fun noValueAreDispatchedFromSecondObservableWhenFirstAsNotCompleted() {
+        firstPublisher.value = "a"
+        secondPublisher.value = "1"
+        firstPublisher.value = "b"
+        assertEquals(listOf("a", "b"), accumulator.values)
+        assertEquals(false, accumulator.completed)
+        assertEquals(null, accumulator.error)
+    }
+
+    @Test
+    fun valuesFromSecondPublisherAreDispatchedAfterFirstPublisherCompleted() {
+        firstPublisher.value = "a"
+        secondPublisher.value = "1"
+        firstPublisher.value = "b"
+        firstPublisher.complete()
+        assertEquals(listOf("a", "b", "1"), accumulator.values)
+        assertEquals(false, accumulator.completed)
+        assertEquals(null, accumulator.error)
+    }
+
+    @Test
+    fun valuesFromSecondPublisherAreBufferedUntilFirstPublisherCompleted() {
```

Comment:
De we really want to accumulate and buffer values on the second "Hot" observable?